temple/utils.py

```diff
@@ -126,3 +126,30 @@ def _call_api(self, verb, url, **request_kwargs):
     def get(self, url, **request_kwargs):
         """Github API get"""
         return self._call_api('get', url, **request_kwargs)
+
+class GitLabClient():
+    def __init__(self, template):
+        api_token = os.environ.get('GITLAB_API_TOKEN')
+        gitlab_api = 'https://gitlab.com/api/v4'
+        self.headers = { 'Authorization': f"Bearer {api_token}" }
+        query = template.split('/')[-1:]
+        url = f"{gitlab_api}/projects?search={query}"
+        search_response = requests.get(url, headers=self.headers)
+        search_result = search_response.json()
+        if len(search_result) != 1:
+            print('Template search inconclusive.')
+            exit()
+        self.file_api = f"{gitlab_api}/projects/{search_result[0]['id']}/repository/files"
+
+    def get(self, void, **request_kwargs):
```

Comment:
There are three usages of `client.get()` but this only handles the second case:
- `/search/code` for `temple ls`
- `/repos/{}/contents/cookiecutter.json` for `_cookiecutter_configs_have_changed`
- `/repos/{}/commits` for `_get_latest_template_version_w_github`